cleverpush/src/main/java/com/cleverpush/database/DatabaseClient.java

```diff
@@ -0,0 +1,32 @@
+package com.cleverpush.database;
+
+import android.content.Context;
+
+import androidx.room.Room;
+
+public class DatabaseClient {
+
+  private Context mCtx;
+  private static DatabaseClient mInstance;
+  private CleverPushDatabase appDatabase;
+
+  private DatabaseClient(Context mCtx) {
+    this.mCtx = mCtx;
+    appDatabase = Room.databaseBuilder(mCtx, CleverPushDatabase.class, "CleverPushDatabase")
```

Comment:
the `DatabaseClient` class is designed to provide a singleton instance of the `CleverPushDatabase` using a specific Context. If we call `DatabaseClient.getInstance(context)` multiple times with different contexts, it will still return the same database instance for a given application.

The `Room.databaseBuilder` method creates or opens a database based on the provided context and database name. If we pass a different context every time and call DatabaseClient.getInstance(context), it will still refer to the same database if the database name ("CleverPushDatabase") and the database version have not changed.